ground/gcs/src/libs/tlmapcontrol/core/providerstrings.cpp

```diff
@@ -67,13 +68,31 @@ QString ProviderStrings::decrypt(QString str)
         ++pos;
     }
 
-    return QString::fromUtf8(qUncompress(ba));
+    QString decryptedString = QString::fromUtf8(qUncompress(ba));
+    if (decryptedString.isEmpty()) {
+        throw std::domain_error(QString("Decrypted to an empty vector!. Input vector: ").append(inputString).toStdString());

```

Comment:
Is there a standard way to make this clear in comments? It seems unfortunate that you had to hunt through the code to find out what the catch was going to be. It sort of is like a `goto` in the sense that it can make it hard to follow the program's progression.